src/app/berkeley_migration_verifier/berkeley_migration_verifier.ml

```diff
@@ -0,0 +1,516 @@
+      |> Test.eval ;
+      Deferred.unit
+
+let get_forked_blockchain uri fork_point =
+  match Caqti_async.connect_pool ~max_size:128 uri with
+  | Error _ ->
+      failwithf "Failed to create Caqti pools for Postgresql to %s"
+        (Uri.to_string uri) ()
+  | Ok pool ->
+      let query_db = Mina_caqti.query pool in
+      query_db ~f:(fun db ->
+          Sql.Berkeley.Block_info.forked_blockchain db fork_point )
+
+let check_forked_chain ~migrated_archive_uri ~fork_state_hash =
+  let%bind forked_blockchain =
+    get_forked_blockchain (Uri.of_string migrated_archive_uri) fork_state_hash
+  in
+
+  match forked_blockchain with
+  | [] ->
+      Deferred.return (Check.skipped "Forked state hash does not exist")
+  | [ _fork_point ] ->
+      Deferred.return (Check.skipped "Forked blockchain is empty")
+  | fork_point :: fork_chain ->
+      Deferred.return
+        ( List.mapi fork_chain ~f:(fun idx block ->
+              let idx_64 = Int64.of_int idx in
+              let protocol_version_check =
+                if Int.( <> ) block.protocol_version_id 2 then
+                  Check.err
+                    (sprintf
+                       "block with id (%d) has unexpected protocol version"
+                       block.id )
+                else Check.ok
+              in
+              let global_slot_since_hardfork_check =
+                if Int64.( = ) block.global_slot_since_hard_fork idx_64 then
+                  Check.err
+                    (sprintf
+                       "block with id (%d) has unexpected \
+                        global_slot_since_hard_fork"
+                       block.id )
+                else Check.ok
+              in
+              let global_slot_since_genesis_check =
+                if
+                  Int64.( > ) block.global_slot_since_genesis
+                    (Int64.( + ) fork_point.global_slot_since_genesis
+                       (Int64.( + ) idx_64 Int64.one) )
+                then
+                  Check.err
+                    (sprintf
+                       "block with id (%d) has unexpected \
+                        global_slot_since_genesis"
+                       block.id )
+                else Check.ok
+              in
+              let block_height_check =
+                if Int.( > ) block.height (fork_point.height + idx + 1) then
+                  Check.err
+                    (sprintf
+                       "block with id (%d) has unexpected \
+                        global_slot_since_genesis"
+                       block.id )
+                else Check.ok
+              in
+
+              Check.combine
+                [ protocol_version_check
+                ; global_slot_since_genesis_check
+                ; global_slot_since_hardfork_check
+                ; block_height_check
+                ] )
+        |> Check.combine )
+
+let compare_migrated_replayer_output ~migrated_replayer_output ~fork_config_file
```

Comment:
Those 2 inputs contains 2 ledgers that both of type `Runtime_config.Ledger.t`. I think it might be better to try to compare them directly within ocaml. Otherwise, it would be better to just run the script directly.